### PR TITLE
GitHub connector with GraphQL implementation

### PR DESCRIPTION
**Pull request by abhishekjoshi-crest**
This PR is a complete GitHub connector with GraphQL approach. Below functionalities are a part of this PR:

Ping
Indexing
End-to-end testing
Advanced Sync rules

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Comment by seanstory** [comment]
With the merge of the github docs from #1119 , this will need to augment those docs to include advanced sync rules and add nuance to the binary content we're fetch (markdown and RST only)

**Comment by abhishekjoshi-crest** [comment]
> With the merge of the github docs from #1119 , this will need to augment those docs to include advanced sync rules and add nuance to the binary content we're fetch (markdown and RST only)

@seanstory, updated the doc related to GitHub server configuration, advanced sync rules, and markdown files.

File: docs/reference/github.md [hunk 1]
```diff
@@ -65,7 +91,15 @@ The connector syncs the following objects and entities:
 
 [Basic sync rules](https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html#sync-rules-basic) are identical for all connectors and are available by default.
 
-Advanced sync rules are not available for this connector in the present version. Currently filtering is controlled by ingest pipelines.
+## Advanced Sync Rules
+
+Advanced Sync Rules are now available in elastic 8.10 version.
```
Comment (seanstory) [inline docs/reference/github.md:96]:
```suggestion
Advanced Sync Rules are available in 8.10+.
```

**Review by seanstory — APPROVED** [review]
one nit, but lgtm

**Review by seanstory — APPROVED** [review]

**Pull request merged**